superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -1287,6 +1296,37 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
     });
   };
 
+  const renderSSHTunnelForm = () => (
+    <SSHTunnelForm
+      isEditMode={isEditMode}
+      sshTunneling={sshTunneling}
+      db={db as DatabaseObject}
```

Comment:
Not new to this PR so maybe not in scope, but is there any validation on `db` to ensure it is a `DatabaseObject`?  Forcing a coercion here and on line 1304 basically removes any useful type checking.   We may want to add a type guard or other validation that db does have the structure SSHTunnelForm expects.

Reply:
Might be wrong but, isn't this:

```
const [db, setDB] = useReducer<
    Reducer<Partial<DatabaseObject> | null, DBReducerActionType>
  >(dbReducer, null);
```
forcing db to be  `Partial<DatabaseObject> | null` ? thus would match the type in `SSHTunnelForm`?

Reply:
@eric-briscoe i say we cut a ticket for this under tech debt so we can revisit doing a proper refactor of this component